lib/msf/ui/console/command_dispatcher/exploit.rb

```diff
@@ -290,17 +290,23 @@ def cmd_rexploit_help
   # Select a reasonable default payload and minimally configure it
   # TODO: Move this somewhere better or make it more dynamic?
   def self.choose_payload(mod)
-    compatible_payloads = mod.compatible_payloads.map(&:first)
-
-    # XXX: This approach is subpar, but at least this list has been reevaluated
-    preferred_payloads = %w[
-      meterpreter/reverse_https
-      meterpreter/reverse_http
-      meterpreter/reverse_tcp
-      shell_reverse_tcp
-      shell/reverse_tcp
-      generic/shell_reverse_tcp
-      generic/shell_bind_tcp
+    compatible_payloads = mod.compatible_payloads(
+      excluded_platforms: ['Multi'] # We don't want to select a multi payload
+    ).map(&:first)
+
+    # If there is only one compatible payload, return it immediately
+    return compatible_payloads.first if compatible_payloads.length == 1
```

Comment:
Just a quick sanity question from me, does this need to execute the same code as the `preferred_payloads` loop does?

> mod.datastore['PAYLOAD'] = payload

Reply:
Good catch. I added this code at the last minute.